From: USB Audio Offload Research <research@audioreach-usb-offload>
Date: Tue, 25 Feb 2026 00:00:00 +0000
Subject: [PATCH 1/6] ASoC: qdsp6: audioreach: add USB audio module IDs and
 interface config

Add MODULE_ID_USB_AUDIO_SINK (0x0700104F) and MODULE_ID_USB_AUDIO_SOURCE
(0x07001050) to audioreach.h. These module IDs are defined in the
AudioReach SPF firmware API (audioreach-graphservices/spf/api/modules/
usb_api.h) and are present in the ADSP firmware on QCS6490/QCM6490
platforms.

Also add PARAM_ID_USB_AUDIO_INTF_CFG (0x080010D6) for configuring the
USB audio interface parameters (sample rate, bit width, channels, device
token, service interval).

These definitions are required for AudioReach (GPR/q6apm) based platforms
to configure USB audio offload graphs through the APM, complementing the
existing APR/q6afe USB offload path.

Signed-off-by: USB Audio Offload Research <research@audioreach-usb-offload>
---
 sound/soc/qcom/qdsp6/audioreach.h | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/sound/soc/qcom/qdsp6/audioreach.h b/sound/soc/qcom/qdsp6/audioreach.h
index XXXXXXX..XXXXXXX 100644
--- a/sound/soc/qcom/qdsp6/audioreach.h
+++ b/sound/soc/qcom/qdsp6/audioreach.h
@@ -33,6 +33,8 @@
 #define MODULE_ID_DISPLAY_PORT_SINK	0x07001069
 #define MODULE_ID_SPEAKER_PROTECTION	0x070010E2
 #define MODULE_ID_SPEAKER_PROTECTION_VI	0x070010E3
+#define MODULE_ID_USB_AUDIO_SINK	0x0700104F
+#define MODULE_ID_USB_AUDIO_SOURCE	0x07001050
 #define MODULE_ID_OPUS_DEC		0x07001174

 #define APM_CMD_GET_SPF_STATE		0x01001021
@@ -150,6 +152,8 @@
 #define PARAM_ID_ENC_BITRATE			0x08001052

+#define PARAM_ID_USB_AUDIO_INTF_CFG		0x080010D6
+
 struct param_id_enc_bitrate_param {
 	uint32_t bitrate;
 } __packed;
@@ -XXX,6 +XXX,24 @@ struct apm_display_port_module_intf_cfg {

 #define APM_DP_INTF_CFG_PSIZE	ALIGN(sizeof(struct apm_display_port_module_intf_cfg), 8)

+/* USB Audio Module Interface Config */
+struct usb_audio_intf_cfg {
+	uint32_t usb_token;
+	uint32_t svc_interval;
+} __packed;
+
+struct apm_usb_module_intf_cfg {
+	struct apm_module_param_data param_data;
+	struct usb_audio_intf_cfg cfg;
+} __packed;
+
+#define APM_USB_INTF_CFG_PSIZE	ALIGN(sizeof(struct apm_usb_module_intf_cfg), 8)
+
 struct audioreach_module_config {
 	int	direction;
 	u32	sample_rate;
@@ -XXX,6 +XXX,8 @@ struct audioreach_module_config {
 	u16	channel_allocation;
 	u16	dp_idx;
+	u32	usb_token;
+	u32	usb_svc_interval;
 };

--
2.43.0
